test/metric_dimension/agent_configs/aggregation_dimensions.json

```diff
@@ -0,0 +1,35 @@
+{
+  "agent": {
+    "metrics_collection_interval": 10
+  },
+  "metrics": {
+    "force_flush_interval": 10,
+    "namespace": "TestAggregationDimensions",
+    "append_dimensions": {
+      "InstanceId": "${aws:InstanceId}",
+      "InstanceType": "${aws:InstanceType}"
+    },
+    "aggregation_dimensions": [
+      [],
+      [
+        "InstanceId"
+      ],
+      [
+        "InstanceId",
+        "InstanceType"
+      ],
+      [
+        "foo",
+        "bar",
+        "InstanceType"
+      ]
+    ],
+    "metrics_collected": {
+      "mem": {
+        "measurement": [
+          "mem_used_percent"
+        ]
```

Comment:
**Nit**:
We could add a append_dimensions here for foo value and eliminate bar too but either way. I'm good with it